extension/src/debugger/languages/dotnet.ts

```diff
@@ -58,98 +57,110 @@ class DotNetService implements IDotNetService {
-                    extensionLogOutputChannel.error(`dotnet build process error: ${err}`);
-                    reject(new Error(buildFailedForProjectWithError(projectFile, err.message)));
-                });
-
-                buildProcess.on('close', (code) => {
-                    if (code === 0) {
-                        // if build succeeds, simply return. otherwise throw to trigger error handling
-                        if (stderrOutput) {
-                            reject(new Error(stderrOutput));
-                        } else {
-                            resolve();
+            if (buildTask) {
+                // Modify the task to target the specific project
+                const projectName = path.basename(projectFile, '.csproj');
+
+                // Create a modified task definition with just the project file
+                const modifiedDefinition = {
+                    ...buildTask.definition,
+                    file: projectFile  // This will make it build the specific project directly
+                };
+
+                // Create a new task with the modified definition
+                const modifiedTask = new vscode.Task(
+                    modifiedDefinition,
+                    buildTask.scope || vscode.TaskScope.Workspace,
+                    `build ${projectName}`,
+                    buildTask.source,
+                    buildTask.execution,
+                    buildTask.problemMatchers
+                );
+
+                extensionLogOutputChannel.info(`Executing build task: ${modifiedTask.name} for project: ${projectFile}`);
+                await vscode.tasks.executeTask(modifiedTask);
+
+                let disposable: vscode.Disposable;
+                return new Promise<void>((resolve, reject) => {
+                    disposable = vscode.tasks.onDidEndTaskProcess(async e => {
+                        if (e.execution.task === modifiedTask) {
+                            if (e.exitCode !== 0) {
+                                reject(new Error(buildFailedWithExitCode(e.exitCode ?? 'unknown')));
+                            }
+                            else {
+                                return resolve();
+                            }
                         }
-                    } else {
-                        reject(new Error(buildFailedForProjectWithError(projectFile, stdoutOutput || stderrOutput || `Exit code ${code}`)));
-                    }
-                });
-            });
+                    });
+                }).finally(() => disposable.dispose());
+            }
+            else {
+                this.writeToDebugConsole(noCsharpBuildTask, 'stdout', true);
+            }
+        }
+        else {
+            this.writeToDebugConsole(csharpDevKitNotInstalled, 'stdout', true);
         }
 
-        // C# Dev Kit may not register the build task immediately, so we need to retry until it is available
-        const pRetry = (await import('p-retry')).default;
```

Comment:
> p-retry is no longer necessary so I removed it.

I'm not familiar with the library, but does it have any advantages over doing the loop ourselves?